enterprise/server/remote_execution/runner/runner.go

```diff
@@ -540,22 +526,23 @@ func (r *commandRunner) cleanupCIRunner(ctx context.Context) error {
 	return res.Error
 }
 
-type ContainerProvider func(ctx context.Context, props *platform.Properties, st *repb.ScheduledTask, state *rnpb.RunnerState, workDir string) (*container.TracedCommandContainer, error)
+type ContainerProvider interface {
+	NewContainer(context.Context, *platform.Properties, *repb.ScheduledTask, *rnpb.RunnerState, string) (container.CommandContainer, error)
+}
```

Comment:
could this go in `enterprise/server/remote_execution/container/container.go` and call it `Provider`? So it's referenced like `container.Provider`